recipes/bx/all/conanfile.py

```diff
@@ -1,81 +1,189 @@
+from conan import ConanFile
+from conan.tools.files import copy, get, rename
+from conan.tools.build import check_min_cppstd
+from conan.tools.layout import basic_layout
+from conan.tools.microsoft import is_msvc
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.microsoft import MSBuild, VCVars
+from conan.tools.gnu import Autotools, AutotoolsToolchain
+from pathlib import Path
 import os
-import glob
-from conans import ConanFile, CMake, tools
-from conans.errors import ConanInvalidConfiguration
 
-class BxConan(ConanFile):
+required_conan_version = ">=1.50.0"
+
+
+class bxConan(ConanFile):
     name = "bx"
-    description = "Base library used across multiple projects."
     license = "BSD-2-Clause"
-    topics = ("conan", "bx")
     homepage = "https://github.com/bkaradzic/bx"
     url = "https://github.com/conan-io/conan-center-index"
-    exports_sources = ["CMakeLists.txt"]
-    generators = "cmake"
-    settings = "os", "arch", "compiler", "build_type"
-    options = {
-        "fPIC": [True, False],
-    }
-    default_options = {
-        "fPIC": True,
-    }
-
-    @property
-    def _source_subfolder(self):
-        return "source_subfolder"
-
-    @property
-    def _build_subfolder(self):
-        return "build_subfolder"
-
-    def config_options(self):
-        if self.settings.os == 'Windows':
-            del self.options.fPIC
+    description = "Base library providing utility functions and macros."
+    topics = ("general", "utility")
+    settings = "os", "compiler", "arch", "build_type"
+    options = {"fPIC": [True, False], "tools": [True, False]}
+    default_options = {"fPIC": True, "tools": False}
+
+    bxFolder = "bx"
+   
+    vsVerToGenie = {"17": "2022", "16": "2019", "15": "2017",
+                    "193": "2022", "192": "2019", "191": "2017"}
+
+    gccOsToGenie = {"Windows": "--gcc=mingw-gcc", "Linux": "--gcc=linux-gcc", "Macos": "--gcc=osx", "Android": "--gcc=android", "iOS": "--gcc=ios"}
+    gmakeOsToProj = {"Windows": "mingw-gcc", "Linux": "linux", "Macos": "osx", "Android": "android", "iOS": "ios"}
+    gmakeArchToGenieSuffix = {"x86": "-x86", "x86_64": "-x64", "armv8": "-arm64", "armv7": "-arm"}
+    osToUseArchConfigSuffix = {"Windows": False, "Linux": False, "Macos": True, "Android": True, "iOS": True}
+
+    buildTypeToMakeConfig = {"Debug": "config=debug", "Release": "config=release"}
+    archToMakeConfigSuffix = {"x86": "32", "x86_64": "64"}
+    osToUseMakeConfigSuffix = {"Windows": True, "Linux": True, "Macos": False, "Android": False, "iOS": False}
```

Comment:
Please, move it to the respective method, there is no re-usage.

Reply:
You are very much correct. Did that now!